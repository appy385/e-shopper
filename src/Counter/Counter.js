import React, { useState } from 'react';
import './Counter.scss';

const Counter = () => {
  const [count, setCount] = useState(0);
  return (
    <div>
      <p>{`count: ${count}`}</p>
      <button type="button" className="counter-button" onClick={() => { setCount(count + 1); }}>Click</button>
    </div>
  );
};
export default Counter;
