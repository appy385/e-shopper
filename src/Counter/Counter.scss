@import '../index.scss';
.counter-button {
  width: 3rem;
  p {
      color: aquamarine;
  }
  @include custom-mixin;
}

@include media-mixin($tablet-breakpoint){
    .counter-button{
        background-color: orange;
    } 
   
}
