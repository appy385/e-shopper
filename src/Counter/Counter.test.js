import {
  fireEvent, render, screen, waitFor,
} from '@testing-library/react';
import React from 'react';
import Counter from './Counter';

describe(Counter.name, () => {
  test('should match snapshot', () => {
    const { container } = render(<Counter />);
    expect(container).toMatchSnapshot();
  });

  test('should display current count', () => {
    render(<Counter />);
    screen.getByText('count: 0');
  });

  test('should display increment button', () => {
    render(<Counter />);
    const incrementElement = screen.getByText('Click');
    expect(incrementElement.tagName).toBe('BUTTON');
  });

  test('should update count to 1', async () => {
    render(<Counter />);
    const incrementElement = screen.getByText('Click');
    fireEvent.click(incrementElement);

    await waitFor(() => screen.getByText('count: 1'));
  });
});
