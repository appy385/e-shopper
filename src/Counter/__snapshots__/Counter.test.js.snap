// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter should match snapshot 1`] = `
<div>
  <div>
    <p>
      count: 0
    </p>
    <button
      class="counter-button"
      type="button"
    >
      Click
    </button>
  </div>
</div>
`;
